add directory itself to keep list

clan opened the pull request:
if not fixed, following sls will delete /root/a/b

``` yaml
 /root/a:
   file.directory:
     - user: root
     - group: root
     - mode: 0755
     - clean: True
     - require:
         - file: /root/a/b

 /root/a/b:
   file.directory:
     - user: root
     - group: root
     - mode: 0755
```

clan committed:
add directory itself to keep list

if not fixed, following sls will delete /root/a/b

 /root/a:
   file.directory:
     - user: root
     - group: root
     - mode: 0755
     - clean: True
     - require:
         - file: /root/a/b

 /root/a/b:
   file.directory:
     - user: root
     - group: root
     - mode: 0755

jfindlay commented:
@clan, thanks for the fix.  In the future, please feel free to submit fixes directly to the oldest, supported release branch applicable, see https://docs.saltstack.com/en/latest/topics/development/contributing.html#which-salt-branch.

clan committed:
another fix for file defined w/ id, but require name

for following sls:

 /root/a:
   file.directory:
     - user: root
     - group: root
     - mode: 0755
     - clean: True
     - require:
         - file: /root/a/b
         - file: /root/a/c

 /root/a/b:
   file.directory:
     - user: root
     - group: root
     - mode: 0755

 id_test:
   file.managed:
     - name: /root/a/c
     - user: root
     - group: root
     - mode: 0755
     - replace: False

clan commented:
ok,  will submit to the oldest branch next time.

clan committed:
check w/ low['name'] only
